services/py-api/src/database/transaction_manager.py

```diff
@@ -55,22 +57,23 @@ async def _retry_tx(func: Callable[..., Awaitable[Result]], *args: Any, **kwargs
         """
         max_retries = 3
         delay = 1  # initial delay in seconds
+        session_id = session.session_id["id"].hex()
 
         # range is exclusive that's why we do max_retries + 1
         for retry in range(1, max_retries + 1):
             try:
-                return await func(*args, **kwargs)
+                return await func(*args, session=session, **kwargs)
```

Comment:
Why the tests didn't catch this? If we have missed adding the session, then how the sequence of operations is aborted?

Reply:
I dont know bro. The session variable did not reach the repository methods. You doubt this? Try it. Remove it and try to log the session from the repository methods.

Reply:
I am not talking about the log; I am talking if the operations in the callback were retried/committed/rollbacked because you don't pass the session explicitly. Either the operations in the callback operate with a `session=None` which should not be the case or the session is passed implicitly via the args and kwargs. We have to check what's the case